src/core/crypto/secp256k1.ts

```diff
@@ -171,7 +174,12 @@ export class Secp256k1PublicKey extends PublicKey {
    * @category Serialization
    */
   static isInstance(publicKey: PublicKey): publicKey is Secp256k1PublicKey {
-    return "key" in publicKey && (publicKey.key as any)?.data?.length === Secp256k1PublicKey.LENGTH;
+    return (
```

Comment:
@gregnazario This needs to be modified; otherwise, the test will fail, because the K1 public key will mistakenly treat R1 as K1, leading to failure.